docs/how-to/instrumenting-rewriting-binary-application.rst

```diff
@@ -0,0 +1,939 @@
+            928  ../examples/lulesh/lulesh.cc              Kokkos::parallel_for<Kokkos::RangePolicy<Kokkos::OpenMP, Kokkos::IndexType<in...         Kokkos::parallel_for<Kokkos::RangePolicy<Kokkos::OpenMP, Kokkos::IndexType<in...
+            960  ../examples/lulesh/lulesh.cc              Kokkos::parallel_for<Kokkos::RangePolicy<Kokkos::OpenMP, Kokkos::IndexType<lo...         Kokkos::parallel_for<Kokkos::RangePolicy<Kokkos::OpenMP, Kokkos::IndexType<lo...
+          21470  ../examples/lulesh/lulesh.cc              LagrangeLeapFrog                                                                         LagrangeLeapFrog(domain) [lulesh.cc]
+           1836  ../examples/lulesh/lulesh.cc              main                                                                                     int main(int, char * *) [lulesh.cc]
+
+Sampling
+========================================
+
+.. note::
+
+   This capability has been deprecated in favor of :doc:`Call stack sampling <./sampling-call-stack>`.
+
+By default, ``omnitrace-instrument`` uses ``--mode trace`` for instrumentation. The ``--mode sampling`` option
+only instruments ``main`` in an executable. It activates both CPU call-stack sampling and
+background system-level thread sampling by default.
+Tracing capabilities which do not rely on instrumentation, such as the HIP API and kernel tracing
+(which is collected by roctracer), are still available.
+
+The Omnitrace sampling capabilities are always available, even in trace mode, but are deactivated by default.
+To activate sampling in trace mode, set ``OMNITRACE_USE_SAMPLING=ON`` in the environment
+or in an Omnitrace configuration file.
+
+Embedding a default configuration
+========================================
+
+Use the ``--env`` option to embed a default configuration into the target. Although this option
+works for runtime instrumentation, it is most useful when generating new binaries because the generated
+binary can be used later on in a different login session when the environment might have changed.
+
+For example, if the following commands are run,
+the configuration settings are not be preserved for subsequent sessions:
+
+.. code-block:: shell
+
+   omnitrace-instrument -o ./foo.inst -- ./foo
+   export OMNITRACE_USE_SAMPLING=ON
+   export OMNITRACE_SAMPLING_FREQ=5
+   omnitrace-run -- ./foo.inst
+
+Whereas the following command preserves those environment variables:
+
+.. code-block:: shell
+
+   omnitrace-instrument -o ./foo.samp --env OMNITRACE_USE_SAMPLING=ON OMNITRACE_SAMPLING_FREQ=5 -- ./foo
+
+They can now be used in future sessions.
+
+.. code-block:: shell
+
+   # will sample 5x per second
+   omnitrace-run -- ./foo.samp
+
+Even though the environment variables are preserved, subsequent sessions can still override those defaults:
+
+.. code-block:: shell
+
+   # will sample 100x per second
+   export OMNITRACE_SAMPLING_FREQ=100
+   omnitrace-run -- ./foo.samp
+
+Troubleshooting
```

Comment:
This is great. 
Maybe these troubleshooting tips can also be referenced or linked to in the "Troubleshooting'-omnitrace.rst" document?